microprofile.jdt/com.redhat.microprofile.jdt.quarkus/src/main/java/com/redhat/microprofile/jdt/internal/quarkus/providers/QuarkusProjectIdentifier.java

```diff
@@ -0,0 +1,20 @@
+package com.redhat.microprofile.jdt.internal.quarkus.providers;
+
+import org.eclipse.jdt.core.IJavaProject;
+import org.eclipse.jdt.core.JavaModelException;
+
+import com.redhat.microprofile.jdt.core.IJavaProjectIdentifier;;
+
+public class QuarkusProjectIdentifier implements IJavaProjectIdentifier {
+
+	@Override
+	public boolean isProject(IJavaProject project) throws JavaModelException {
+		return project!= null && project.findType("io.quarkus.runtime.Application") != null;
+	}
+
+	@Override
+	public String getProjectLabel() {
+		// TODO Auto-generated method stub
+		return "QUARKUS";
```

Comment:
When a label is static, generally it's the extension point declaration which declares it (with a label XML attribute likeyou have class attribute) and not the class.